engine/flink/management/periodic/src/test/scala/pl/touk/nussknacker/engine/management/periodic/UtilsSpec.scala

```diff
@@ -2,32 +2,104 @@ package pl.touk.nussknacker.engine.management.periodic
 
-class UtilsSpec extends TestKit(ActorSystem("UtilsSpec")) with AnyWordSpecLike with Matchers with BeforeAndAfterAll {
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+
+class UtilsSpec
+    extends TestKit(ActorSystem("UtilsSpec"))
+    with AnyWordSpecLike
+    with Matchers
+    with BeforeAndAfterAll
+    with LazyLogging {
 
   override def afterAll(): Unit = {
     TestKit.shutdownActorSystem(system)
   }
 
+  class TestActor extends Actor {
+    override def receive: Receive = { case _ => () }
+  }
+
   "Utils" should {
 
-    "gracefully stop actor and free actor path" in {
-      class TestActor extends Actor {
-        override def receive: Receive = { case _ =>
-          ()
+    "create an actor if it's name is free" in {
+      import system.dispatcher
+
+      val actorName = "actorName1" // unique name in each test so that they don't interfere with each other
+
+      createActorWithRetry(actorName, Props(new TestActor), system)
+    }
+
+    "create an actor if it's name isn't free but is freed before retrying gives up - idle actor" in {
+      import system.dispatcher
+
+      val actorName = "actorName2" // unique name in each test so that they don't interfere with each other
+
+      val actorRef = createActorWithRetry(actorName, Props(new TestActor), system)
+
+      val futureA = Future {
+        createActorWithRetry(actorName, Props(new TestActor), system)
+      }
+
+      val futureB = Future {
+        Thread.sleep(1000)
+        system.stop(actorRef)
+      }
+
+      Await.result(Future.sequence(Seq(futureA, futureB)), Duration.Inf)
+    }
+
+    "create an actor if it's name isn't free but is freed before retrying gives up - busy actor" in {
+      class BusyTestActor extends Actor {
+        override def receive: Receive = { case msg =>
+          logger.info(s"Sleeping on the job '$msg' ...")
+          Thread.sleep(1000)
         }
       }
-      val actorName = "actorName"
 
-      val actorRef = system.actorOf(Props(new TestActor), actorName)
+      import system.dispatcher
+
+      val actorName = "actorName3" // unique name in each test so that they don't interfere with each other
+
+      val actorRef = createActorWithRetry(actorName, Props(new BusyTestActor), system)
+
+      var messageCounter = 0
+      while (messageCounter < 1000) {
+        actorRef ! s"message number $messageCounter"
+        messageCounter += 1
+      }
+
+      val futureA = Future {
+        createActorWithRetry(actorName, Props(new BusyTestActor), system)
+      }
+
+      val futureB = Future {
+        Thread.sleep(1000)
+
+        // if this was gracefulStop, it would take too long to stop the actor, as it would continue processing it's messages
+        system.stop(actorRef)
+      }
+
+      Await.result(Future.sequence(Seq(futureA, futureB)), Duration.Inf)
+    }
+
+    "fail to create an actor if it's name isn't freed" in {
```

Comment:
we could change the behaviour to also call `system.stop(actor)` if it's taken, this way `def createActorWithRetry` would always succeed, but perhaps that's more powerful than what's needed, as it's only ever called after the actors are told to stop by `toClose` anyway